src/composition/watchers/watcher/subgraph.rs

```diff
@@ -84,123 +71,28 @@ impl SubgraphWatcherKind {
             Self::Introspect(introspection) => introspection.watch(),
             // TODO: figure out what this is; sdl? stdin one-off? either way, probs not watching
             Self::_Once(_) => unimplemented!(),
-
-            // Create a new single buffered channel for testing watch events.
-            #[cfg(test)]
-            Self::TestWatcher => {
-                use tokio::sync::mpsc::channel;
-                use tokio_stream::wrappers::ReceiverStream;
-
-                let (tx, rx) = channel(1);
-                tx.send("watch event".to_string()).await.unwrap();
-                ReceiverStream::new(rx).boxed()
-            }
         }
     }
 }
 
-/// Subgraph introspection
-#[derive(Debug, Clone)]
-pub struct SubgraphIntrospection {
-    endpoint: SubgraphUrl,
-    // TODO: ticket using a hashmap, not a tuple, in introspect opts as eventual cleanup
-    headers: Option<Vec<(String, String)>>,
-}
-
-//TODO: impl retry (needed at least for dev)
-impl SubgraphIntrospection {
-    fn new(endpoint: SubgraphUrl, headers: Option<Vec<(String, String)>>) -> Self {
-        Self { endpoint, headers }
-    }
-
-    // TODO: better typing so that it's over some impl, not string; makes all watch() fns require
-    // returning a string
-    fn watch(&self) -> Pin<Box<dyn Stream<Item = String> + Send>> {
-        let client = reqwest::Client::new();
-        let endpoint = self.endpoint.clone();
-        let headers = self.headers.clone();
-
-        let (tx, rx) = unbounded_channel();
-        let rx_stream = UnboundedReceiverStream::new(rx);
-
-        // Spawn a tokio task in the background to watch for subgraph changes
-        tokio::spawn(async move {
-            // TODO: handle errors?
-            let _ = SubgraphIntrospect {
-                opts: IntrospectOpts {
-                    endpoint,
-                    headers,
-                    watch: true,
-                },
-            }
-            .run(
-                client,
-                &OutputOpts {
-                    format_kind: RoverOutputFormatKind::default(),
-                    output_file: None,
-                    // Attach a transmitter to stream back any subgraph changes
-                    channel: Some(tx),
-                },
-                // TODO: impl retries (at least for dev from cli flag)
-                None,
-            )
-            .await;
-        });
-
-        // Stream any subgraph changes, filtering out empty responses (None) while passing along
-        // the sdl changes
-        rx_stream
-            .filter_map(|change| async move {
-                match change {
-                    OutputChannelKind::Sdl(sdl) => Some(sdl),
-                }
-            })
-            .boxed()
-    }
-}
-
 /// A unit struct denoting a change to a subgraph, used by composition to know whether to
 /// recompose.
-pub struct SubgraphChanged;
+pub struct SdlChanged {
```

Comment:
Any chance we could call this `SubgraphSchemaChanged` to match the design doc? It's kind of a small nit, but I'm hoping to keep the language unified, if possible.

Reply:
changed